### PR TITLE
Feature add layer editform

### PR DESCRIPTION
**Pull request by hvostya**
tfs#109192

### CODE REVIEW

**Comment by kuzkok** [comment]
You shouldn't import ember-flexberry translation, they are not changes in this addon.
In dummy locales must be 2 imports - one for this addon, and one for ember-flexberry

File: addon/models/new-platform-flexberry-g-i-s-map-layer.js [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+
+  @class NewPlatformFlexberryGISMapLayerModel
+  @extends BaseModel
+*/
 let Model = BaseModel.extend({
   name: DS.attr('string'),
   type: DS.attr('string'),
   visibility: DS.attr('boolean'),
-  settings: DS.attr('jsonobject'),
+  settings: DS.attr('string'),
```
Comment (kuzkok) [inline addon/models/new-platform-flexberry-g-i-s-map-layer.js]: need to remove jsonobject transform files too

File: addon/locales/ru/translations.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import Ember from 'ember';
+import emberFlexberryTranslations from 'ember-flexberry/locales/ru/translations';
+
+const translations = {};
+Ember.$.extend(true, translations, emberFlexberryTranslations);
+
+Ember.$.extend(true, translations, {'forms': {
```
Comment (kuzkok) [inline addon/locales/ru/translations.js]: need line break before 'forms'

File: addon/models/new-platform-flexberry-g-i-s-map-layer.js (hunk 1 above)
Comment (hvostya) [inline addon/models/new-platform-flexberry-g-i-s-map-layer.js]: I'll fix it.

File: addon/components/new-platform-flexberry-g-i-s-map-layer-setting.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  Component for editing of layer settings that are kept as string but must be edited as object.
+
+  @example
+  In order to add component following template can be used:
+  ```handlebars
+  {{new-platform-flexberry-g-i-s-map-layer-setting
+    value=model.settingsAsObject
+    linkToFieldValue=model.type
+    renderInto='new-platform-flexberry-g-i-s-map-layer-edit'
+    saveValueToFieldName='settings'
```
Comment (SeregaMatin) [inline addon/components/new-platform-flexberry-g-i-s-map-layer-setting.js]: We use double quotes in templates, so in .hbs examples we shell use them too.

File: addon/models/new-platform-flexberry-g-i-s-map-layer-tile.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    minZoom: { numericality: { allowBlank: true, onlyInteger: true, greaterThanOrEqualTo: 0 } },
+    maxZoom: { numericality: { allowBlank: true, onlyInteger: true, greaterThanOrEqualTo: 0 } },
+    maxNativeZoom: { numericality: { allowBlank: true, onlyInteger: true, greaterThanOrEqualTo: 0 } },
+    tileSize: { numericality: { allowBlank: true, onlyInteger: true, greaterThanOrEqualTo: 0 } },
+    zoomOffset: { numericality: { allowBlank: true, onlyInteger: true, greaterThanOrEqualTo: 0 } },
+    opacity: { numericality: { allowBlank: true, greaterThanOrEqualTo: 0 } }
+  }
+});
+
+export default Model;
```
Comment (SeregaMatin) [inline addon/models/new-platform-flexberry-g-i-s-map-layer-tile.js]: Any projectons?

File: addon/models/new-platform-flexberry-g-i-s-map-layer-wms.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  transparent: DS.attr('boolean'),
+  version: DS.attr('string'),
+  crs: DS.attr('string'),
+
+  validations: {
+    url: { presence: true }
+  }
+});
+
+export default Model;
```
Comment (SeregaMatin) [inline addon/models/new-platform-flexberry-g-i-s-map-layer-wms.js]: Same question. Any projectons?

File: app/templates/new-platform-flexberry-g-i-s-map-layer-edit.hbs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    }}
+    {{flexberry-validationmessage error=model.errors.type pointing="pointing"}}
+  </div>
+  <div class="ui segment">
+    {{new-platform-flexberry-g-i-s-map-layer-setting
+      value=model.settingsAsObject
+      linkToFieldValue=model.type
+      renderInto='new-platform-flexberry-g-i-s-map-layer-edit'
+      saveValueToFieldName='settings'
+    }}
```
Comment (SeregaMatin) [inline app/templates/new-platform-flexberry-g-i-s-map-layer-edit.hbs]: Are any validation messages possible for model.settingsAsObject?

File: addon/models/new-platform-flexberry-g-i-s-map-layer-tile.js (hunk 4 above)
Comment (hvostya) [inline addon/models/new-platform-flexberry-g-i-s-map-layer-tile.js]: No need. It is not simple model. It doesn't have serializer, it won't be saved to database.

File: addon/models/new-platform-flexberry-g-i-s-map-layer-wms.js (hunk 5 above)
Comment (hvostya) [inline addon/models/new-platform-flexberry-g-i-s-map-layer-wms.js]: No need. It is not simple models. It doesn't have serializer, it won't be saved to database.

File: app/templates/new-platform-flexberry-g-i-s-map-layer-edit.hbs (hunk 6 above)
Comment (hvostya) [inline app/templates/new-platform-flexberry-g-i-s-map-layer-edit.hbs]: I do not understand.
Template of "new-platform-flexberry-g-i-s-map-layer-setting" displays validation inside.
What validation do you wand for model.settingsAsObject?

File: addon/models/new-platform-flexberry-g-i-s-map-layer-tile.js (hunk 4 above)
Comment (SeregaMatin) [inline addon/models/new-platform-flexberry-g-i-s-map-layer-tile.js]: Ok.

File: addon/models/new-platform-flexberry-g-i-s-map-layer-wms.js (hunk 5 above)
Comment (SeregaMatin) [inline addon/models/new-platform-flexberry-g-i-s-map-layer-wms.js]: I see, thaks.

File: addon/components/new-platform-flexberry-g-i-s-map-layer-setting.js (hunk 3 above)
Comment (hvostya) [inline addon/components/new-platform-flexberry-g-i-s-map-layer-setting.js]: Fixed.

**Pull request merged**